### PR TITLE
Add an option for encoding scalars with newlines to Emitter.Options

### PR DESCRIPTION
**Pull request by tejassharma96**
Solves #405 by adding a new option for encoding scalars which contain newlines

### CODE REVIEW

**Comment by acecilia** [comment]
Thanks! Looking forward to this. @jpsim wdyt? 🙏

**Comment by danibachar** [comment]
+1 for this too

**Review by acecilia — APPROVED** [review]

File: Yams.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name                      = 'Yams'
-  s.version                   = '5.0.6'
+  s.version                   = '5.1.0'
```
Comment (danibachar) [inline Yams.podspec:3]: Version should be updated to 5.2.0

Comment (jpsim) [reply Yams.podspec:3]: Please don’t touch this file. I’ll bump the version when the time comes to cut a release.

**Review by jpsim — APPROVED** [review]
This looks good! Could you please add a change log entry and rebase? Thanks for your contribution.

**Comment by tejassharma96** [comment]
ok rebased and updated the changelog 🙏

**Comment by tejassharma96** [comment]
older swift versions do not understand `any Protocol` and the tests are failing because of that:
```
Encoder.swift:229:45: protocol 'StringProtocol' can only be used as a generic constraint because it has Self or associated type requirements
```

updated to add a `#if swift(>=5.6)`, which I _believe_ is when support for that was added, and for prior versions just try to cast to `String` instead of `any StringProtocol`

**Comment by jpsim** [comment]
@tejassharma96 just making sure you see the CI failures here.

**Comment by tejassharma96** [comment]
yes, sorry - somewhat distracted. I don't have an environment where I can install old versions of xcode since I'm on sonoma :/ how would you suggest I debug?

**Comment by tejassharma96** [comment]
Ah actually - based on [this article](https://www.donnywals.com/what-is-the-any-keyword-in-swift/), the check should be for 5.7 instead of 5.6:
>  in Swift 5.7 the requirements around protocols with a Self requirement have been relaxed

Updated the check, could you see if CI succeeds now please?

**Comment by tejassharma96** [comment]
hmmm, one failure still but looks unrelated to this change:
```
Failing tests:
	YamsTests:
		PerformanceTests.testUsingComposeWithUTF8()
		PerformanceTests.testUsingComposeWithUTF8()
```

will see if I can repro locally

**Comment by jpsim** [comment]
CI is all ✅ 

I’ll merge shortly.

**Comment by acecilia** [comment]
Thank you both 🙌

**Pull request merged**